runtime/doc/lua.txt

```diff
@@ -4432,4 +4432,175 @@ tohtml.tohtml({winid}, {opt})                         *tohtml.tohtml.tohtml()*
         (`string[]`)
 
 
+==============================================================================
+Lua module: vim.net                                                  *vim.net*
+
+*HeaderTable*
+
+    Fields: ~
+      • {set}      (`fun(self: HeaderTableself: HeaderTablevalue: string[]|stringkey: string)`)
+                   Set value of header.
+      • {append}   (`fun(self: HeaderTableself: HeaderTablekey: string)`)
+                   Append value to header.
+      • {get}      (`fun(self: HeaderTableself: HeaderTablekey: string): string?`)
+                   Get header value. For headers like cookies, use
+                   |HeaderTable:get_all()|.
+      • {get_all}  (`fun(self: HeaderTableself: HeaderTablekey: string): string[]?`)
+                   Get all header values.
+      • {has}      (`fun(self: HeaderTableself: HeaderTablekey: string): boolean`)
+
+*vim.net.createCurlArgs.Opts*
+
+    Fields: ~
+      • {method}             (`vim.net.HttpMethod?`)
+      • {redirect}           (`"follow"|"error"|"none"?`)
+      • {data}               (`string|table<string, any>?`)
+      • {headers}            (`HeaderTable|table<string, string|string[]>?`)
+      • {download_location}  (`string?`)
+      • {upload_file}        (`string?`)
+      • {credentials}        (`string?`)
+
+
+HeaderTable:append({key})                               *HeaderTable:append()*
+    Append value to header.
```

Comment:
I'm really not a fan of the vim.opt-style custom functions for basic table operations. Could this not be a normal table instead? IIRC even python's requests package use a basic dictionary for providing headers.